backend/src/crd/controller/viewer/reconciler/reconciler.go

```diff
@@ -83,6 +83,16 @@ func (r *Reconciler) Reconcile(req reconcile.Request) (reconcile.Result, error)
 	}
 	glog.Infof("Got instance: %+v", view)
 
+	if len(view.Spec.TensorboardSpec.TensorflowImage) == 0 {
```

Comment:
the viewer crd is not tensorflow specific. it's good to move tensorboard specific code to a single place or abstract that out. 

```
switch view.Spec.Type :
 viewerV1beta1.ViewerTypeTensorboard :
     tensorboard_handler()
 any_other_type: 
      {
		glog.Infof("Unsupported spec type: %q", view.Spec.Type)
		// Return nil to indicate nothing more to do here.
		return reconcile.Result{}, nil
	}

...
tensorboard_handler(){
  if  len(view.Spec.TensorboardSpec.TensorflowImage) == 0 {...}
}
```

Reply:
So far we don't support other type https://github.com/kubeflow/pipelines/blob/d5e27e291fbe09cfa63a9de0f5b4fdea65af7c6f/backend/src/crd/controller/viewer/reconciler/reconciler.go#L87

And I feel most of the code is tensorflow specific. For this time, I plan to put the specific code after the line (pasted above) and say it's sufficient for now. Later, if we have other types, we probably will separate most of the current logic from the added type support.